File: web_api/schemas/sop.py
from pydantic import BaseModel


class SopRecord(BaseModel):
    id: str
    shop_id: str
    domain: str
    title: str
    version: str
    status: str
    indexed: bool
    updated_at: str


class SopDetail(SopRecord):
    content: str
    published_at: str | None = None


class SopCreate(BaseModel):
    shop_id: str
    domain: str
    title: str
    content: str
    version: str = "draft"
    status: str = "draft"


class SopUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    version: str | None = None
    status: str | None = None

File: web_frontend/src/api/sop.ts
import { apiGet } from "./client";

export type SopRecord = {
  id: string;
  shop_id: string;
  domain: string;
  title: string;
  version: string;
  status: string;
  indexed: boolean;
  updated_at: string;
};

export type SopDetail = SopRecord & {
  content: string;
  published_at: string | null;
};

export function getSopRecords() {
  return apiGet<{ items: SopRecord[]; total: number }>("/api/sop");
}

export function getSopDetail(id: string) {
  return apiGet<SopDetail>(`/api/sop/${id}`);
}
